File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第4章奖学金.py
#奖学金  lanqiaoOJ 531
count = int(input())
info = []
for i in range(count):
    info.append([i+1] + list(map(int,input().split())))
for i in info:
    i.append(sum(i)-info.index(i)-1)
index = reversed((4,1,0))
for i in index:
    info.sort(key=lambda x:x[i],reverse=True)
for i in range(0,5):
    print(info[i][0],info[i][4])

File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第4章奖学金.cpp
//奖学金  lanqiaoOJ 531 
#include<bits/stdc++.h>
using namespace std;
struct stu{
    int id;      //学号
    int c,m,e;   //语、数、外
    int sum;
}st[305];
bool cmp(stu a,stu b){
    if(a.sum > b.sum)       return true;
    else if(a.sum < b.sum)  return false;
    else{             //a.sum == b.sum
        if(a.c > b.c)       return true;
        else if(a.c < b.c)  return false;
        else{         //a.c == b.c
            if(a.id > b.id) return false;
            else return true;
        }
    }
}
int main(){
    int n;    cin>>n;
    for(int i=1;i<=n;i++){
        st[i].id = i;                              //学号
        cin >> st[i].c >> st[i].m >> st[i].e;
        st[i].sum = st[i].c + st[i].m + st[i].e;   //总分
    }
    sort(st+1,st+1+n,cmp);
    for(int i=1;i<=5;i++)  cout<<st[i].id<<' '<<st[i].sum<<"\n";
    return 0;
}
